alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportAcknowledgeAlertAction.kt

```diff
@@ -112,48 +128,85 @@ class TransportAcknowledgeAlertAction @Inject constructor(
                                     .optionalTimeField(Alert.ACKNOWLEDGED_TIME_FIELD, Instant.now())
                                     .endObject()
                             )
-                    )
-                } else {
-                    emptyList()
+                        updateRequests.add(updateRequest)
+                    } else {
+                        val copyRequest = IndexRequest(AlertIndices.ALERT_HISTORY_WRITE_INDEX)
+                            .routing(request.monitorId)
+                            .id(alert.id)
+                            .source(
+                                alert.copy(state = Alert.State.ACKNOWLEDGED, acknowledgedTime = Instant.now())
+                                    .toXContentWithUser(XContentFactory.jsonBuilder())
+                            )
+                        copyRequests.add(copyRequest)
+                    }
                 }
             }
 
-            log.info("Acknowledging monitor: $request.monitorId, alerts: ${updateRequests.map { it.id() }}")
-            val bulkRequest = BulkRequest().add(updateRequests).setRefreshPolicy(request.refreshPolicy)
-            client.bulk(
-                bulkRequest,
-                object : ActionListener<BulkResponse> {
-                    override fun onResponse(response: BulkResponse) {
-                        onBulkResponse(response)
-                    }
-
-                    override fun onFailure(t: Exception) {
-                        actionListener.onFailure(AlertingException.wrap(t))
-                    }
-                }
-            )
+            try {
+                val updateResponse = if (updateRequests.isNotEmpty())
+                    client.bulk(BulkRequest().add(updateRequests).setRefreshPolicy(request.refreshPolicy)).actionGet()
```

Comment:
can we not make this asynchronous?

Reply:
No, as we need this for the output for the `AcknowledgeResponse`